bin/build_homework_function.sh

```diff
@@ -13,78 +12,77 @@ function clean_homework {
+    if [ -d "$hw_build_path/custom_validation_code/" ]; then
+	rsync -ruz --delete "$hw_build_path/custom_validation_code/"    "$course_dir/custom_validation_code/$assignment/"
     fi
 
     # change permissions so other instructor users in this course & DAEMON_USER can re-run the build course script
-    find $course_dir/build/                   -type d -exec chmod -f ug+rwx,g+s,o= {} \;
-    find $course_dir/build/                   -type f -exec chmod -f ug+rw,o= {} \;
-    find $course_dir/build/                           -exec chgrp -f ${course_group} {} \;
-    find $course_dir/provided_code/           -type d -exec chmod -f ug+rwx,g+s,o= {} \;
-    find $course_dir/provided_code/           -type f -exec chmod -f ug+rw,o= {} \;
-    find $course_dir/provided_code/                   -exec chgrp -f ${course_group} {} \;
-    find $course_dir/instructor_solution/           -type d -exec chmod -f ug+rwx,g+s,o= {} \;
-    find $course_dir/instructor_solution/           -type f -exec chmod -f ug+rw,o= {} \;
-    find $course_dir/instructor_solution/                   -exec chgrp -f ${course_group} {} \;
-    find $course_dir/generated_output/           -type d -exec chmod -f ug+rwx,g+s,o= {} \;
-    find $course_dir/generated_output/           -type f -exec chmod -f ug+rw,o= {} \;
-    find $course_dir/generated_output/                   -exec chgrp -f ${course_group} {} \;
-    find $course_dir/test_input/              -type d -exec chmod -f ug+rwx,g+s,o= {} \;
-    find $course_dir/test_input/              -type f -exec chmod -f ug+rw,o= {} \;
-    find $course_dir/test_input/                      -exec chgrp -f ${course_group} {} \;
-    find $course_dir/test_output/             -type d -exec chmod -f ug+rwx,g+s,o= {} \;
-    find $course_dir/test_output/             -type f -exec chmod -f ug+rw,o= {} \;
-    find $course_dir/test_output/                     -exec chgrp -f ${course_group} {} \;
-    find $course_dir/custom_validation_code/  -type d -exec chmod -f ug+rwx,g+s,o= {} \;
-    find $course_dir/custom_validation_code/  -type f -exec chmod -f ug+rw,o= {} \;
-    find $course_dir/custom_validation_code/          -exec chgrp -f ${course_group} {} \;
-    find $course_dir/config/build/            -type d -exec chmod -f ug+rwx,g+s,o= {} \;
-    find $course_dir/config/build/            -type f -exec chmod -f ug+rw,o= {} \;
-    find $course_dir/config/build/            -exec chgrp -f ${course_group} {} \;
-    find $course_dir/config/complete_config/  -type d -exec chmod -f ug+rwx,g+s,o= {} \;
-    find $course_dir/config/complete_config/  -type f -exec chmod -f ug+rw,o= {} \;
-    find $course_dir/config/complete_config/  -exec chgrp -f ${course_group} {} \;
+    find "$course_dir/build/"                   -type d -exec chmod -f ug+rwx,g+s,o= {} \;
+    find "$course_dir/build/"                   -type f -exec chmod -f ug+rw,o= {} \;
+    find "$course_dir/build/"                           -exec chgrp -f ${course_group} {} \;
```

Comment:
Might be good to also quote ${course_group}. Also for all other instances in this file.